test/services/active_challenges_cache.js

```diff
@@ -0,0 +1,186 @@
+      activeChallengesCache.expire('0xdeadface');
+
+      expect(activeChallengesCache.activeChallenges).to.deep.include.members([
+        aChallenge(),
+        aChallengeWith(anotherChallengeId)
+      ]);
+    });
+  });
+
+  describe('when decreasing active count', () => {
+    it('reduces count for the right challenge', () => {
+      activeChallengesCache.add(aChallengeWith({challengeId: '0xc0ffee', count: 12}));
+      activeChallengesCache.add(aChallengeWith({challengeId: '0xbeef', count: 43}));
+
+      activeChallengesCache.decreaseActiveCount('0xc0ffee');
+
+      expect(activeChallengesCache.activeChallenges).to.deep.include.members([
+        aChallengeWith({challengeId: '0xc0ffee', count: 11}),
+        aChallengeWith({challengeId: '0xbeef', count: 43})
+      ]);
+    });
+
+    it('reduces count for same challenge many times', () => {
+      activeChallengesCache.add(aChallengeWith({challengeId: '0xc0ffee', count: 12}));
+
+      activeChallengesCache.decreaseActiveCount('0xc0ffee');
+      activeChallengesCache.decreaseActiveCount('0xc0ffee');
+      activeChallengesCache.decreaseActiveCount('0xc0ffee');
+
+      expect(activeChallengesCache.activeChallenges).to.deep.equal([aChallengeWith({challengeId: '0xc0ffee', count: 9})]);
+    });
+
+    it('does not throw if tried to decrease active count of not existing challenge', () => {
+      activeChallengesCache.add(aChallengeWith({challengeId: '0xc0ffee'}));
+
+      activeChallengesCache.decreaseActiveCount([{challengeId: '0xdeadface'}]);
+      expect(activeChallengesCache.activeChallenges).to.deep.include(aChallengeWith({challengeId: '0xc0ffee'}));
+    });
+
+    it('removes challenge when active count falls down to 0', () => {
+      activeChallengesCache.add(aChallengeWith({challengeId: '0xc0ffee', count: 1}));
+
+      activeChallengesCache.decreaseActiveCount('0xc0ffee');
+
+      expect(activeChallengesCache.activeChallenges).not.to.deep.include(aChallengeWith({challengeId: '0xc0ffee'}));
+    });
+  });
+
+  it('returns active challenges as a sorted array', () => {
+    activeChallengesCache.add(aChallengeWith({challengeId: '0xbaddad', blockNumber: 4}));
+    activeChallengesCache.add(aChallengeWith({challengeId: '0xc0ffee', blockNumber: 1}));
+    activeChallengesCache.add(aChallengeWith({challengeId: '0xbeef', blockNumber: 3}));
+    activeChallengesCache.add(aChallengeWith({challengeId: '0xdeadface', blockNumber: 2, logIndex: 1}));
+    activeChallengesCache.add(aChallengeWith({challengeId: '0xcafebabe', blockNumber: 2, logIndex: 0}));
+
+    expect(activeChallengesCache.activeChallenges).to.deep.equal([
+      aChallengeWith({challengeId: '0xc0ffee', blockNumber: 1}),
+      aChallengeWith({challengeId: '0xcafebabe', blockNumber: 2, logIndex: 0}),
+      aChallengeWith({challengeId: '0xdeadface', blockNumber: 2, logIndex: 1}),
+      aChallengeWith({challengeId: '0xbeef', blockNumber: 3}),
+      aChallengeWith({challengeId: '0xbaddad', blockNumber: 4})
```

Comment:
IMO: there is a lot of repetition here with all the `aChallenge...` calls. Maybe assign them to local variables? This will make the method a little bit longer, but IMO more readable.